src/Http/AssetsApiClientFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Drupal\mux\Http;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ImmutableConfig;
+use GuzzleHttp\Client;
+use MuxPhp\Api\AssetsApi;
+use MuxPhp\Configuration;
+
+/**
+ * Service for generating Mux Assets API clients.
+ */
+class AssetsApiClientFactory {
+
+  /**
+   * The config settings object.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  protected ImmutableConfig $config;
+
+  /**
+   * Constructs a new AssetsApiClientFactory instance.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory service.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->config = $config_factory->get('mux.settings');
```

Comment:
Config Name könntest du noch auslagern, falls sich der mal ändern sollte.

Reply:
Wohin würdest du den auslagern? Nur im File oder Modul-weit?

Reply:
Ja in eine Konstante der Klasse.
Du brauchst den Namen auch noch im Setter. 
Detail...